.github/workflows/ci.yaml

```diff
@@ -0,0 +1,128 @@
+                           k, v, default_mark)
+                     print(epilog, end="")
+                 except KeyError:
+                     sys.exit(
+                       "** Default version is undefined **\n> {}".format(proton))
+                 if not default_is_valid:
+                     sys.exit(
+                       "** Default version is invalid **\n> {}".format(proton))
+        shell: python
+
+  download_files:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: '3.x'
+      - name: Test download_files(), including HTTP redirection handling
+        run: |
+             import importlib
+             import os
+             import shutil
+             import sys
+             import tempfile
+             shutil.move("truckersmp-cli", "main.py")
+             sys.path.append(os.getcwd())
+             main = importlib.import_module("main")
+             basedir = "/lhark/truckersmp-cli/raw/f67b887b25747f5601549b8cbe2a7e8f9fa64ce7/"
+             files = [
+               ("Makefile", "af9fa17e5a5d03a703b363b7d9712f5d"),
+               ("LICENSE", "702573e867acc76789f7a8c8c861e8fb"),
+             ]
+             files_to_download = []
+             with tempfile.TemporaryDirectory() as tempdir:
+                 for name, md5 in files:
+                     files_to_download.append(
+                       (os.path.join(basedir, name),
+                        os.path.join(tempdir, name),
+                        md5))
+                 main.download_files("github.com", files_to_download)
+        shell: python
+
+  flake8:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: '3.x'
+      - name: Install flake8, flake8-docstrings, and flake8-import-order
+        run: pip install flake8 flake8-docstrings flake8-import-order
+      - name: Check main script by flake8
+        run: flake8 --max-line-length=90 --show-source --statistics truckersmp-cli
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install gcc-mingw-w64-x86-64
+        run: |
+             sudo apt -q update
+             sudo apt -q install --no-install-recommends gcc-mingw-w64-x86-64
+      - name: Build truckersmp-cli.exe
+        run: make clean && make
+      - name: Make output files
+        run: |
+             mkdir $OUTDIR
+             echo "Making non-stripped binary..."
+             fakeroot tar Jcvf \
+               $OUTDIR/${{ github.event_name }}-${{ github.sha }}.tar.xz ./*
+             sha256sum *.exe
+             x86_64-w64-mingw32-strip *.exe
+             echo -e "\nMaking stripped binary..."
+             fakeroot tar Jcvf \
+               $OUTDIR/${{ github.event_name }}-${{ github.sha }}-stripped.tar.xz ./*
+             sha256sum *.exe
```

Comment:
* You're not _making_ the files here, you're only compressing and stripping.
* Where is the sha256sum-file coming from?
* Why even compressing one (binary) file?
* Do we even need to strip the binary, an executable which is (unstripped!) < 400 kB?

Reply:
1. Changed to "Prepare files for uploading".
2. *I'm thinking we'll use auto-built `truckersmp-cli.exe` (from the "Artifact") in the future (when we need to change `truckersmp-cli.c`)*. I think checksums will be useful to identify.
3. It's possible to specify "output" directory for "Artifact" but the resulted file will be zip format and "executable" attribute of our main script (`truckersmp-cli`) will not be preserved. Now this is outdated because only `truckersmp-cli.exe` and stripped file will be uploaded in zip format.
4. Same reason as "2.". In addition, the size of stripped file is only about 20 kB. `truckersmp-cli.c` is a very small program.

Reply:
> * Where is the sha256sum-file coming from?

Sry, I've misread the line.

> 2\. _I'm thinking we'll use auto-built `truckersmp-cli.exe` (from the "Artifact") in the future (when we need to change `truckersmp-cli.c`)_. I think checksums will be useful to identify.

Let's keep this just for debugging and verification purposes. We need to know if it builds and whether it does what it should do. For this we need to be able to access the binary and give it a quick run to verify it's working correctly locally.
Isn't the commit id enough for identification?

> 4\. the size of stripped file is only about 20 kB.

We definitely don't need both versions here. I'm more about the stripping itself. Since this is for verification and debugging purposes stripping is against that. But… is debugging with gdb something we would do anyway?
Let's keep it stripped here. If we ever have to debug the binary we have to change it.